### DISCUSSION TITLE
How can I obtain the total wire length?

### DISCUSSION THREAD

**Discussion by huang-kaixin** [The-OpenROAD-Project/OpenROAD-flow-scripts#1491] [Q&A] [answered]
Is there a separate command to report the total wirelength after Detail routing? If available, how is it calculated? In which directory is the source code located? Thank you for your response!

**Comment by maliberty** [comment]
@osamahammad21 ?

**Comment by osamahammad21** [comment, marked as answer]
@huang-kaixin There is no separate command for getting the total wirelength. You should be able to find a report of the total wire length on each layer and all layers accumulated at the end of detailed routing if verbosity level > 0.

You could find the source code responsible for this under src/drt/src/dr/FlexDR.cpp -> void FlexDR::end(bool done)

https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/drt/src/dr/FlexDR.cpp#L766-L779
